main/tests.py

```diff
@@ -76,4 +78,59 @@ def test_dog_owner_homepage_is_visible_for_dog_owner(self, client, create_dog_ow
         client.force_login(user=create_dog_owner_user.user)
         response = client.get("/homepage/")
         assert response.status_code == 200
-        assert list(response.context['daycares']) == list(DayCare.objects.all())
+        assert set(response.context['day_care_queryset']) == set(DayCare.objects.all())
+
+
+@pytest.mark.django_db
+class TestDogOwnerHomePageView:
+    def test_search_daycare_present_only_available_daycares_on_specific_dates(self, client, create_dog_owner_user):
+        test_day_care = DayCare.create(username='testuser', email='test@email.com', password='test_password',
+                                       name='name', description='test description', price_per_day=100,
+                                       capacity=1, area='CENTER', city='tel aviv', address='address')
+        client.force_login(user=create_dog_owner_user.user)
+        search_form = {'area': "",
+                       'city': "",
+                       'price_per_day': 100,
+                       'name': "",
+                       'start_date': "2022-05-03",
+                       'end_date': "2022-05-08",
+                       }
+        response = client.post('/homepage/', search_form, follow=True)
+        day_care_queryset = response.context['day_care_queryset']
+        assert test_day_care in day_care_queryset
+        Order.create(dog_owner_id=DogOwner.objects.get(id=1), daycare_id=test_day_care,
```

Comment:
belong in a different test

Reply:
Can you explain ? Do you think that create day-care with capcity=0, and test that it doesn't appear in the queryset is better approch ?

Reply:
on the first assertion you check that a daycare with capacity is available, 
in the second one you check  that a daycare with no capacity is not available. 
those are two different cases.
BUT, on second thought, I think it can keep it like this. but put spaces between different parts of the test.

Reply:
and change the name of the test. it doesn't check that the daycare is available  on specific dates. it checks that a daycare is not available after its capacity been *reduced* to 0.

Reply:
You right thanks for making it clear.

Reply:
Fixed